test/box-luatest/gh_6530_net_box_opts_check_test.lua

```diff
@@ -0,0 +1,142 @@
+local server = require('test.luatest_helpers.server')
+local netbox = require('net.box')
+local t = require('luatest')
+local g = t.group()
+
+g.before_all = function()
+    g.server = server:new{alias = 'default'}
+    g.server:start()
+    g.server:exec(function()
+        local s = box.schema.space.create('test')
+        s:format({
+                  {name = 'id',   type = 'unsigned'},
+                  {name = 'data', type = 'string'},
+                 })
+        s:create_index('primary', {parts = {'id'}})
+        box.schema.user.grant('guest', 'read,write', 'space', 'test')
+    end)
+end
+
+g.after_all = function()
+    g.server:exec(function()
+        if box.space.test then
+            box.space.test:drop()
+        end
+    end)
+    if g.conn then
+        g.conn:close()
+    end
+    g.server:drop()
+end
+
+g.test_netbox_connect_opts = function()
+    t.assert_error_msg_contains("unexpected option 'some_opt'", function()
+        netbox:connect(g.server.net_box_uri, {some_opt = 1})
+    end)
+    t.assert_error_msg_contains("parameter 'user' should be of type string", function()
+        netbox:connect(g.server.net_box_uri, {user = true})
+    end)
+    t.assert_error_msg_contains("parameter 'password' should be of type string", function()
+        netbox:connect(g.server.net_box_uri, {password = true})
+    end)
+    t.assert_error_msg_contains("parameter 'wait_connected' should be one of types: number, boolean", function()
+        netbox:connect(g.server.net_box_uri, {wait_connected = 'string'})
+    end)
+    t.assert_error_msg_contains("parameter 'reconnect_after' should be of type number", function()
+        netbox:connect(g.server.net_box_uri, {reconnect_after = true})
+    end)
+    t.assert_error_msg_contains("parameter 'call_16' should be of type boolean", function()
+        netbox:connect(g.server.net_box_uri, {call_16 = 1})
+    end)
+    t.assert_error_msg_contains("parameter 'console' should be of type boolean", function()
+        netbox:connect(g.server.net_box_uri, {console = 1})
+    end)
+    t.assert_error_msg_contains("parameter 'connect_timeout' should be of type number", function()
+        netbox:connect(g.server.net_box_uri, {connect_timeout = 'string'})
+    end)
+    t.assert_error_msg_contains("parameter 'required_protocol_version' should be of type number", function()
+        netbox:connect(g.server.net_box_uri, {required_protocol_version = 'string'})
+    end)
+    t.assert_error_msg_contains("parameter 'required_protocol_features' should be of type table", function()
+        netbox:connect(g.server.net_box_uri, {required_protocol_features = 1})
+    end)
+
+    g.conn = netbox.connect(g.server.net_box_uri, {wait_connected = 1})
+    t.assert(g.conn)
```

Comment:
You shouldn't rely on test case execution order. I.e. you can't assume that the next test case will be executed after this line and you will have `g.conn` set. If you want to keep these test cases separated into functions, then you can make them regular local functions, take `conn` as an argument, and call them in this test case like

```Lua
test_netbox_ping_opts(conn)
test_netbox_all_methods_opts(conn)
```